app/src/main/java/de/westnordost/streetcomplete/quests/barrier_type/AddBarrierType.kt

```diff
@@ -14,7 +14,7 @@ import de.westnordost.streetcomplete.osm.Tags
 class AddBarrierType : OsmFilterQuestType<BarrierType>() {
 
     override val elementFilter = """
-        nodes with barrier = yes
+        nodes, ways with barrier = yes
```

Comment:
note that it is deliberate: `barrier=yes` line may be fence, wall, retaining wall... And retaining walls are directional, so supporting them would require supporting way reversing which is not implemented and tricky to implement.

Reply:
Changed it.